pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -185,13 +190,18 @@ func initIdentity(set *apps.StatefulSet, pod *v1.Pod) {
 	// Set these immutable fields only on initial Pod creation, not updates.
 	pod.Spec.Hostname = pod.Name
 	pod.Spec.Subdomain = set.Spec.ServiceName
+	if pod.Labels == nil {
+		pod.Labels = make(map[string]string)
+	}
+
 }
 
-// updateIdentity updates pod's name, hostname, and subdomain to conform to set's name and headless service.
+// updateIdentity updates pod's name, hostname, and subdomain, and StatefulSetPodNameLabel to conform to set's name
+// and headless service.
 func updateIdentity(set *apps.StatefulSet, pod *v1.Pod) {
 	pod.Name = getPodName(set, getOrdinal(pod))
 	pod.Namespace = set.Namespace
-
+	pod.Labels[StatefulSetPodNameLabel] = pod.Name
```

Comment:
Is it possible that `pod.Labels` has become `nil` again at this point? Do we enforce anywhere that Pods must have at least one label?

Reply:
it shouldn't but it doesn't hurt to check